src/solvers/dgmulti/dg_parabolic.jl

```diff
@@ -0,0 +1,275 @@
+function create_cache(mesh::DGMultiMesh, equations::AbstractParabolicEquations,
+                      dg::DGMultiWeakForm, RealT, uEltype)
+  nvars = nvariables(equations)
+
+  @unpack M, Drst = dg.basis
+  weak_differentiation_matrices = map(A -> -M \ (A' * M), Drst)
+
+  # u_transformed stores "transformed" variables for computing the gradient
+  @unpack md = mesh
+  u_transformed = allocate_nested_array(uEltype, nvars, size(md.x), dg)
+  u_grad = ntuple(_ -> similar(u_transformed), ndims(mesh))
+  viscous_flux = similar.(u_grad)
+
+  u_face_values = allocate_nested_array(uEltype, nvars, size(md.xf), dg)
+  scalar_flux_face_values = similar(u_face_values)
+  grad_u_face_values = ntuple(_ -> similar(u_face_values), ndims(mesh))
+
+  local_u_values_threaded = [similar(u_transformed, dg.basis.Nq) for _ in 1:Threads.nthreads()]
+  local_viscous_flux_threaded = [ntuple(_ -> similar(u_transformed, dg.basis.Nq), ndims(mesh)) for _ in 1:Threads.nthreads()]
+  local_flux_face_values_threaded = [similar(scalar_flux_face_values[:, 1]) for _ in 1:Threads.nthreads()]
+
+  return (; u_transformed, u_grad, viscous_flux,
+            weak_differentiation_matrices,
+            u_face_values, grad_u_face_values, scalar_flux_face_values,
+            local_u_values_threaded, local_viscous_flux_threaded, local_flux_face_values_threaded)
+end
+
+# Transform solution variables prior to taking the gradient
+# (e.g., conservative to primitive variables). Defaults to doing nothing.
+# TODO: can we avoid copying data?
+function transform_variables!(u_transformed, u, equations)
+  @threaded for i in eachindex(u)
+    u_transformed[i] = u[i]
+  end
+end
+
+# interpolates from solution coefficients to face quadrature points
+function prolong2interfaces!(u_face_values, u, mesh::DGMultiMesh, equations::AbstractParabolicEquations,
```

Comment:
In #378 we always used `u, gradients` instead of `gradients, u` when passing both solution and gradients to a function, e.g., in https://github.com/trixi-framework/Trixi.jl/pull/378/files#diff-7e9c6178128fe81c398396fe444b955c231ae55ef42a84a8a748a7f53bf23770R11-R12. IIRC, we generally have the order "general argument, specific argument", or am just so out of sync that it is actually the other way around 😬? cc @ranocha

Reply:
The other way round: https://trixi-framework.github.io/Trixi.jl/stable/styleguide/ :sweat_smile: 
In this case, I wouldn't see `gradients` as a specification of `u`, though. However, we might discuss whether we want to adopt the conventions of other packages, e.g., the SciML stuff where derivatives come before the solution itself (e.g., https://diffeq.sciml.ai/stable/types/dynamical_types/). In case we want to do this, the order suggested by @jlchan is already fine.

Reply:
Thanks for pointing me to the docs 🙏 No, then I agree - let's keep it the way @jlchan suggested. However, I think I favor something more descriptive (maybe `gradients` or `gradients_u`), but this discussion should go into #1147 (already added).

Resolving this now.